idea/idea-gradle/src/org/jetbrains/kotlin/idea/configuration/GradleKotlinFrameworkSupportProvider.kt

```diff
@@ -187,21 +187,59 @@ open class GradleKotlinJavaFrameworkSupportProvider(
     override fun getDescription() = "A Kotlin library or application targeting the JVM"
 }
 
-open class GradleKotlinJSFrameworkSupportProvider(
-    frameworkTypeId: String = "KOTLIN_JS",
-    displayName: String = "Kotlin/JS"
+abstract class GradleKotlinJSFrameworkSupportProvider(
+    frameworkTypeId: String,
+    displayName: String
 ) : GradleKotlinFrameworkSupportProvider(frameworkTypeId, displayName, KotlinIcons.JS) {
+    abstract val jsSubTargetName: String
+
+    override fun addSupport(
+        buildScriptData: BuildScriptDataBuilder,
+        module: Module,
+        sdk: Sdk?,
+        specifyPluginVersionIfNeeded: Boolean,
+        explicitPluginVersion: String?
+    ) {
+        super.addSupport(buildScriptData, module, sdk, specifyPluginVersionIfNeeded, explicitPluginVersion)
+
+        buildScriptData.addOther(
+            """
+            kotlin {
+                target {
+                    $jsSubTargetName {
+                    }
+                }
+            }""".trimIndent()
+        )
+    }
 
     override fun getPluginId() = KotlinJsGradleModuleConfigurator.KOTLIN_JS
     override fun getPluginExpression() = "id 'org.jetbrains.kotlin.js'"
-
     override fun getDependencies(sdk: Sdk?) = listOf(MAVEN_JS_STDLIB_ID)
-
     override fun getTestDependencies() = listOf(MAVEN_JS_TEST_ID)
-
     override fun getDescription() = "A Kotlin library or application targeting JavaScript"
 }
 
+open class GradleKotlinJSBrowserFrameworkSupportProvider(
+    frameworkTypeId: String = "KOTLIN_JS_BROWSER",
+    displayName: String = "Kotlin/JS for browser"
+) : GradleKotlinJSFrameworkSupportProvider(frameworkTypeId, displayName) {
+    override val jsSubTargetName: String
+        get() = "browser"
+
+    override fun getDescription() = "A Kotlin library or application targeting JavaScript for browser"
+}
+
+open class GradleKotlinJSNodeFrameworkSupportProvider(
+    frameworkTypeId: String = "KOTLIN_JS_NODE",
+    displayName: String = "Kotlin/JS for nodejs"
+) : GradleKotlinJSFrameworkSupportProvider(frameworkTypeId, displayName) {
+    override val jsSubTargetName: String
+        get() = "nodejs"
+
+    override fun getDescription() = "A Kotlin library or application targeting JavaScript for nodejs"
```

Comment:
Minor: maybe stylize nodejs as Node.js (the official stylization)?

Reply:
Indeed. Will do.

Reply:
Fixed